### ISSUE TITLE
hasOne join issues

### ISSUE THREAD

**Issue by LennartOlsen** [go-pg/pg#263]
Hi, I am seeing an issue in regards to the hasOne logic when getting my models through the ORM layer.

My models are as follows:
```go
type Entity struct {
	ID string						'valid:"UUID" json:"id"'
	CompanyName string 				'valid:"length(1|255)" json:"company_name"'
	TradingName string 				'valid:"length(1|255)" json:"trading_name"'

	EntityPerson *EntityPerson 		'valid:"json" json:"entity_person,omitempty"'
}

type EntityPerson struct {
	ID string						'valid:"UUID" json:"id"'
	EntityID string					'valid:"-" json:"entity_id"'
}
```

And my model is as follows:

```go
e := Entity{ID: id}

if err := s.conn.Model(&e).
	Column("entity.*", "EntityPerson").
	Limit(10).
	Select(); err != nil {
	return Entity{}, err
}
return e, nil
```

This is my PGsql structure:

```sql
CREATE TABLE entities (
    id character varying(255) DEFAULT (NOT NULL::boolean) NOT NULL,
    company_name character varying(255) DEFAULT 'NOT NULL'::character varying,
    trading_name character varying(255) DEFAULT 'NOT NULL'::character varying
);

CREATE TABLE entity_persons (
    id character varying(255) NOT NULL,
    entity_id character varying(255)
);
```

By the logic of has one it would seem that i would be able to create a relation from entity to entity_person but when i try to run it get the following error:

```
ERROR #42703 column "EntityPerson" does not exist
```

I have the sql dump turned on as well:
```sql
SELECT "entity".*, "EntityPerson" FROM "entities" AS "entity" LIMIT 10
```

I am using `gopkg.in/pg.v4` and go version `go1.6.2`

**Comment by vmihailenco** [comment]
AFAIK this is belongs to relation and I just added support for it in v4.7.0. Please upgrade and let me know how it works.

**Comment by LennartOlsen** [comment]
Works beautifully with a 0:1 relation, but 0:n i worse

if i change my model to :
```go
type Entity struct {
    ID string                       'valid:"UUID" json:"id"'
    CompanyName string              'valid:"length(1|255)" json:"company_name"'
    TradingName string              'valid:"length(1|255)" json:"trading_name"'

    EntityPerson []*EntityPerson      'valid:"json" json:"entity_person,omitempty"'
}

type EntityPerson struct {
    ID string                       'valid:"UUID" json:"id"'
    EntityID string                 'valid:"-" json:"entity_id"'
}
```

**Comment by vmihailenco** [comment]
Yeah, 0:n is not yet supported. I will take a look a bit later.

**Comment by vmihailenco** [comment]
@LennartOlsen should be fixed in v4.7.1
